server/src/workspaceLookup.ts

```diff
@@ -42,7 +40,7 @@ export class WorkspaceLookup {
     init(): Promise<void> {
         this.workspace.onDidChangeWorkspaceFolders(event => {
             // This event is fired when the user adds or removes a folder to/from their workspace
-            this.collectionPromise(event.added, this.addFolder)
+            event.added.map(this.addFolder);
```

Comment:
Shouldn't these promises be `await`ed? Or is it fine if they'll eventually probably happen?

Reply:
so what happened here is that I used `collectionPromise` on multiple places in the code, and then actually checked how it resolved or rejected as a single void result.

it should also be fine that the scanning of folders runs in the background, it is very unlikely the user has a file open that refers to something in a newly added folder. If they do, the only thing they have to do is switch to another document and back and the processing of the workspace folder will likely be done.

I don't think the signature of `onDidChangeWorkspaceFolders` supports waiting on a promise.